core/storage/classifier/gae_models.py

```diff
@@ -12,28 +12,32 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""Models for storing the classification models."""
+"""Models for storing the classification data models."""
 
 from core.platform import models
+
+import feconf
 import utils
 
 from google.appengine.ext import ndb
 
 (base_models,) = models.Registry.import_models([models.NAMES.base_model])
 
 # Available choices of algorithms for classification.
-ALGORITHM_CHOICES = [
-    'LDAStringClassifier'
-]
+ALGORITHM_CHOICES = []
+for classifier_details in (
+        feconf.INTERACTION_CLASSIFIER_MAPPING.values()):
+    ALGORITHM_CHOICES.append(classifier_details['algorithm_id'])
 
 
-class ClassifierModel(base_models.BaseModel):
+class ClassifierDataModel(base_models.BaseModel):
     """Storage model for classifier used for answer classification.
 
     The id of instances of this class has the form
     {{exp_id}}.{{random_hash_of_16_chars}}
     """
-
+    # The job_request_id received from the TrainClassifierJobModel
+    job_request_id = ndb.StringProperty(required=True, indexed=True)
```

Comment:
`job_request_id` is not a separate field. It is the ID of the instance. So `self.id = job_request_id`. During tests you might not have `job_request_id` so you can use any alphanumeric string. Once this is done we don't have to need `_generate_id()` function because we will specify IDs explicitly when creating the instance. Does that clear your doubt?